Enum member access in if condition causes parser error

## Description

Using an enum member access expression (e.g., `Color.Red`) directly in an if condition causes a parser error. Storing the enum value in a variable first works correctly.

## Reproduction

```ez
const Color enum { Red, Green, Blue }

do main() {
    temp c Color = Color.Red

    // This FAILS with parser error:
    if c == Color.Red {
        println("red")
    }
}
```

## Error

```
error[E2002]: expected :, got ( instead
```

## Expected Behavior

Enum member access should work in all expression contexts, including if conditions.

## Note

This works correctly:
```ez
temp red Color = Color.Red
if c == red {
    println("red")
}
```

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -2639,10 +2639,17 @@ func (p *Parser) parseMemberExpression(left Expression) Expression {
 	// Only if member starts with uppercase (type naming convention) AND is not all uppercase.
 	// All-uppercase names like STATUS.ACTIVE are enum members, not struct types.
 	// Struct types use PascalCase (e.g., Person, MyType).
+	// Additionally, if the object (left side) starts with uppercase, it's likely an enum
+	// access (e.g., Color.Red) not a module.TypeName struct literal.
 	memberName := exp.Member.Value
 	if p.peekTokenMatches(LBRACE) && len(memberName) > 0 && memberName[0] >= 'A' && memberName[0] <= 'Z' && !isAllUpperCase(memberName) {
 		// Build the qualified name from the member expression
 		if ident, ok := left.(*Label); ok {
+			// If the object (module name) also starts with uppercase, this is likely
+			// an enum access like Color.Red, not a struct literal like module.Person{}
+			if len(ident.Value) > 0 && ident.Value[0] >= 'A' && ident.Value[0] <= 'Z' {
+				return exp // It's an enum access, not a struct literal
+			}
 			qualifiedName := &Label{
 				Token: exp.Token,
 				Value: ident.Value + "." + memberName,

--- a/pkg/parser/parser_test.go
+++ b/pkg/parser/parser_test.go
@@ -2073,3 +2073,27 @@ func TestReservedEnumName(t *testing.T) {
 		t.Error("expected error for reserved enum name")
 	}
 }
+
+func TestEnumMemberInIfCondition(t *testing.T) {
+	// Test: enum member access in if condition should parse correctly
+	// Fixes #339
+	input := `const Color enum { Red, Green, Blue }
+do main() {
+    temp c Color = Color.Red
+    if c == Color.Red {
+        println("red")
+    }
+}`
+	l := NewLexer(input)
+	p := NewWithSource(l, input, "test.ez")
+	program := p.ParseProgram()
+
+	if p.EZErrors().HasErrors() {
+		t.Errorf("unexpected parser errors: %v", p.Errors())
+	}
+
+	// Should have parsed without errors
+	if program == nil {
+		t.Fatal("program is nil")
+	}
+}

